x-pack/solutions/security/plugins/security_solution/docs/testing/test_plans/detection_response/prebuilt_rules/prebuilt_rule_import.md

```diff
@@ -91,195 +119,429 @@ User stories:
-And isCustomized should be true
+Then the rule should be updated
+And the updated rule should be prebuilt
+And the updated rule should be marked as non-customized
+And the updated rule's parameters should match the import payload
 ```
 
-#### Scenario: Importing a custom rule with a matching custom `rule_id` and `version`
+### Importing single prebuilt customized rules
 
-**Automation**: 1 cypress test and 1 integration test.
+#### **Scenario: Importing a customized rule when it's not installed**
+
+**Automation**: 1 API integration test.
 
 ```Gherkin
-Given the import payload contains a custom rule
-And its rule_id and version match a rule asset from the installed package
-And the overwrite flag is set to true
+Given the import payload contains a prebuilt rule
+And this rule has a base version in the installed package
+And this rule is different from its base version
+And this rule is not installed
 When the user imports the rule
-Then the rule should be created or updated
-And the ruleSource type should be "internal"
+Then the rule should be created
+And the created rule should be prebuilt
+And the created rule should be marked as customized
+And the created rule's parameters should match the import payload
 ```
 
-#### Scenario: Importing a prebuilt rule with a matching `rule_id` but no matching `version`
+#### **Scenario: Importing a customized rule on top of an installed non-customized rule**
 
-**Automation**: 1 integration test.
+**Automation**: 1 API integration test.
 
 ```Gherkin
 Given the import payload contains a prebuilt rule
-And its rule_id matches one or a few rule assets from the installed package
-And its version does NOT match any of those rule assets
+And this rule has a base version in the installed package
+And this rule is different from its base version
+And this rule is already installed and is not customized
 When the user imports the rule
-Then the rule should be created or updated
-And the ruleSource type should be "external"
-And isCustomized should be true
+Then the rule should be updated
+And the updated rule should be prebuilt
+And the updated rule should be marked as customized
+And the updated rule's parameters should match the import payload
 ```
 
-#### Scenario: Importing a prebuilt rule with a non-existent `rule_id`
+#### **Scenario: Importing a customized rule on top of an installed customized rule**
 
-**Automation**: 1 integration test.
+**Automation**: 1 API integration test.
 
 ```Gherkin
 Given the import payload contains a prebuilt rule
+And this rule has a base version in the installed package
+And this rule is different from its base version
+And this rule is already installed and is customized
+When the user imports the rule
+Then the rule should be updated
+And the updated rule should be prebuilt
+And the updated rule should be marked as customized
+And the updated rule's parameters should match the import payload
+```
+
+### Importing single custom rules
```

Comment:
Do we want to have these cases in the `prebuilt_rule_import` test plan? I guess we're explicitly testing against the package and for the `rule_source` field but these don't pertain to prebuilt rules otherwise

Reply:
@dplumlee Great question. When I just started rewriting this test plan, my first reaction was to remove scenarios for custom rules. But then I realized that we should also explicitly test that with the new logic, custom rules are still being imported as custom rules in terms of their `immutable` and `rule_source` field values. That's what these scenarios intend to cover.

In this:

```Gherkin
Then the rule should be created
And the created rule should be custom
And the created rule's parameters should match the import payload
```

the `created rule should be custom` means that it has `immutable == false` and `rule_source == { type: 'internal' }`, which should be covered by dedicated tests.